tests/phpunit/tests/user/queryCache.php

```diff
@@ -0,0 +1,706 @@
+				'args' => array( 'fields' => array( 'display_name' ) ),
+			),
+			'invalid_field'         => array(
+				'args' => array( 'fields' => array( 'invalid_field' ) ),
+			),
+			'valid array inc id'    => array(
+				'args' => array( 'fields' => array( 'display_name', 'user_email', 'id' ) ),
+			),
+			'valid array inc ID'    => array(
+				'args' => array( 'fields' => array( 'display_name', 'user_email', 'ID' ) ),
+			),
+			'partly valid array'    => array(
+				'args' => array( 'fields' => array( 'display_name', 'invalid_field' ) ),
+			),
+			'orderby'               => array(
+				'args' => array(
+					'fields'  => array( 'ID' ),
+					'orderby' => array( 'login', 'nicename' ),
+				),
+			),
+			'meta query'            => array(
+				'args' => array(
+					'fields'     => array( 'ID' ),
+					'meta_query' => array(
+						'foo_key' => array(
+							'key'     => 'foo',
+							'compare' => 'EXISTS',
+						),
+					),
+					'orderby'    => 'foo_key',
+					'order'      => 'DESC',
+				),
+			),
+			'meta query LIKE'       => array(
+				'args' => array(
+					'fields'     => array( 'ID' ),
+					'meta_query' => array(
+						array(
+							'key'     => 'foo',
+							'value'   => '00',
+							'compare' => 'LIKE',
+						),
+					),
+					'orderby'    => 'foo_key',
+					'order'      => 'DESC',
+				),
+			),
+			'published posts'       => array(
+				'args' => array(
+					'has_published_posts' => true,
+					'fields'              => array( 'ID' ),
+				),
+			),
+			'published posts order' => array(
+				'args' => array(
+					'orderby' => 'post_count',
+					'fields'  => array( 'ID' ),
+				),
+			),
+			'published count_total' => array(
+				'args' => array(
+
+					'count_total' => false,
+					'fields'      => array( 'ID' ),
+				),
+			),
+			'capability'            => array(
+				'args' => array(
+					'capability' => 'install_plugins',
+					'fields'     => array( 'ID' ),
+				),
+			),
+			'include'               => array(
+				'args' => array(
+					'includes' => self::$author_ids,
+					'fields'   => array( 'ID' ),
+				),
+			),
+			'exclude'               => array(
+				'args' => array(
+					'exclude' => self::$author_ids,
+					'fields'  => array( 'ID' ),
+				),
+			),
+			'search'                => array(
+				'args' => array(
+					'search' => 'User',
+					'fields' => array( 'ID' ),
+				),
+			),
+		);
+
+		if ( is_multisite() ) {
+			$data['spam']    = array(
+				'args' => array( 'fields' => array( 'spam' ) ),
+			);
+			$data['deleted'] = array(
+				'args' => array( 'fields' => array( 'deleted' ) ),
+			);
+		}
+
+		return $data;
+	}
+
+	/**
+	 * @ticket 40613
+	 */
+	public function test_query_cache_remove_user_role() {
+		$user_id = self::factory()->user->create( array( 'role' => 'author' ) );
+
+		$q1 = new WP_User_Query(
+			array(
+				'role' => 'author',
+			)
+		);
+
+		$found = wp_list_pluck( $q1->get_results(), 'ID' );
+
+		$this->assertContains( $user_id, $found, 'Find author in returned values' );
```

Comment:
```suggestion
		$this->assertContains( $user_id, $found, 'Expected to find author in returned values.' );
```

:1234: This needs to be applied in several tests below too.